src/client/ducks/bfstats/selectors/index.js

```diff
@@ -5,25 +5,36 @@ import _ from 'lodash'
 export type { StatusType } from 'ducks/fetch/selectors'
 
 export const getBuildsEntities = (state: Object) =>
-  _.get(state, ['entities', 'builds'], {})
+  _.get(state, ['entities', 'builds'], null)
 
 export const getSourceStampsEntities = (state: Object) =>
-  _.get(state, ['entities', 'sourceStamps'], {})
+  _.get(state, ['entities', 'sourceStamps'], null)
 
 export const getRevision = (state: Object, props: Object) =>
-  (props ? props.revision : null)
+  (props && props.revision ? props.revision : null)
 
 export const getSourceStamp = createSelector(
   getSourceStampsEntities, getRevision,
-  (sourceStamps, revision) => (sourceStamps && revision ? sourceStamps[revision] : null)
+  (sourceStamps, revision) => (
+    sourceStamps && revision && sourceStamps.hasOwnProperty(revision)
+      ? sourceStamps[revision] : null)
 )
 
 export const getBuilds = createSelector(
   getBuildsEntities, getSourceStamp,
   (builds, sourceStamp) => {
     if (!builds || !sourceStamp) {
-      return []
+      return null
     }
-    return sourceStamp.nodes.map(id => builds[id])
+
+    return sourceStamp.latestBuilds.nodes.map(id => builds[id])
+  }
+)
+
+export const getABVBuild = createSelector(
+  getBuilds,
+  builds => {
+    const index = _.findIndex(builds, build => build.builder.name === 'proj0-ABuildVerification')
```

Comment:
Ah, then I definitely think we should move over to builds as due to test instabilities there may be 3 ABV builds on the latest changeset with two being FAILED and one being SUCCESS, and we'd like to see all of them, even if the order is: fail, success, fail.
Forgot that latestBuilds already filter and group, though. :)